Row.h

```diff
@@ -0,0 +1,218 @@
+#ifndef HEADER_Row
+#define HEADER_Row
+/*
+htop - Row.h
+(C) 2004-2015 Hisham H. Muhammad
+(C) 2023 htop dev team
+Released under the GNU GPLv2+, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <sys/types.h>
+
+#include "Object.h"
+#include "RichString.h"
+#include "ProcessField.h"
+
+
+typedef enum ProcessField_ {
+   NULL_PROCESSFIELD = 0,
+   PID = 1,
+   COMM = 2,
+   STATE = 3,
+   PPID = 4,
+   PGRP = 5,
+   SESSION = 6,
+   TTY = 7,
+   TPGID = 8,
+   MINFLT = 10,
+   MAJFLT = 12,
+   PRIORITY = 18,
+   NICE = 19,
+   STARTTIME = 21,
+   PROCESSOR = 38,
+   M_VIRT = 39,
+   M_RESIDENT = 40,
+   ST_UID = 46,
+   PERCENT_CPU = 47,
+   PERCENT_MEM = 48,
+   USER = 49,
+   TIME = 50,
+   NLWP = 51,
+   TGID = 52,
+   PERCENT_NORM_CPU = 53,
+   ELAPSED = 54,
+   SCHEDULERPOLICY = 55,
+   PROC_COMM = 124,
+   PROC_EXE = 125,
+   CWD = 126,
+
+   /* Platform specific fields, defined in ${platform}/ProcessField.h */
+   PLATFORM_PROCESS_FIELDS
+
+   /* Do not add new fields after this entry (dynamic entries follow) */
+   LAST_PROCESSFIELD
+} ProcessField;
+
+/* Extend ProcessField with dynamic entries - i.e. defined at runtime */
+#define ROW_DYNAMIC_FIELDS LAST_PROCESSFIELD
+typedef enum ProcessField_ RowField;
+
+extern uint8_t Row_fieldWidths[LAST_PROCESSFIELD];
+#define ROW_MIN_PID_DIGITS 5
+#define ROW_MAX_PID_DIGITS 19
+#define ROW_MIN_UID_DIGITS 5
+#define ROW_MAX_UID_DIGITS 20
+extern int Row_pidDigits;
+extern int Row_uidDigits;
+
+struct Table_;
+struct Machine_;
+struct Settings_;
+
+/* Class representing entities (such as processes) that can be
+ * represented in a tabular form in the lower half of the htop
+ * display. */
+
+typedef struct Row_ {
+   /* Super object for emulated OOP */
+   Object super;
+
+   /* Pointer to quasi-global data */
+   const struct Machine_* host;
+
+   int id;
+   uid_t uid;
+   int group;
```

Comment:
Ah, k. I think for the sake of simplicity™ of this PR, we can leave them there for now.

Future work: Keep uid/gid as part of the `Process`, but provide an interface from `Row` to control the associated behaviour. In particular for the group handling this could be mostly loosely mirroring the `gid` field onto `group` (not equal, as things like "applications" and other concepts could hijack the group field for advanced usage.

For the `uid` field, the highlighting aspect would probably be better suited as its own API (for passing display attributes like background, new, dead, …) too; mostly triggered from the Process class instead.

Reply:
@natoscott 
How about you add a virtual function `IsHighlighted` on the `Row` class so that the `uid` member be moved to `Process`?
`Process_IsHighlighted` would then control whether a data row would be highlighted.